src/Microsoft.Extensions.Diagnostics.HealthChecks/HealthCheck.cs

```diff
@@ -11,7 +11,7 @@ namespace Microsoft.Extensions.Diagnostics.HealthChecks
     /// A simple implementation of <see cref="IHealthCheck"/> which uses a provided delegate to
     /// implement the check.
     /// </summary>
-    public class HealthCheck : IHealthCheck
+    public sealed class HealthCheck : IHealthCheck
```

Comment:
Since when have we used 'sealed' anywhere?

Reply:
Only good reason is perf and devirtualisation. In that case if needed (probably not here) make a private sealed version internally of the public unsealed. Keeps it friendly and you get virtualisation. Public sealed types are nasty.

Reply:
There's no place for virtualization here, since this is basically just a holder of a name-func pair. If people want to avoid `sealed` here though I'm not opposed.